### PR TITLE
feat(repeat): deprecate immediately on job scheduler

### PR DESCRIPTION
**Pull request by manast**
This PR set the immediately option in the job scheduler as deprecated. Immediately has been a hack in its introduction, to mitigate the way "every" works, which in some cases can result in an undesired behaviour if the "every" setting is a large amount of time, lets say one month, the jobs could not be produced until many days or in worst case scenario 29 days had passed. With this PR however, immediately becomes the standard. This is a small breaking change that we hope will not affect any users, specially as this only affects the recently added "upsertJobScheduler". I think it is a good trade-off instead of having to wait for a major version.

### CODE REVIEW

File: tests/test_job_scheduler.ts [hunk 1]
```diff
@@ -744,11 +747,11 @@ describe('Job Scheduler', function () {
       { data: { foo: 'bar' } },
```
Comment (roggervalf) [inline tests/test_job_scheduler.ts:747]: in previous expression, immediately option can be removed as it's not used anymore

File: tests/test_job_scheduler.ts [hunk 2]
```diff
@@ -744,11 +747,11 @@ describe('Job Scheduler', function () {
...
     );
 
-    this.clock.tick(100);
+    this.clock.tick(nextTick);
```
Comment (roggervalf) [inline tests/test_job_scheduler.ts:750]: we can keep this 100 milliseconds as nextTick is 2 seconds. To test that first execution is processed immediately is better to have a first delay lower than 2 seconds

File: src/classes/job-scheduler.ts [hunk 3]
```diff
@@ -69,7 +69,6 @@ export class JobScheduler extends QueueBase {
     }
 
     const prevMillis = opts.prevMillis || 0;
-    now = prevMillis < now ? now : prevMillis;
 
     // Check if we have a start date for the repeatable job
     const { startDate } = repeatOpts;
```
Comment (roggervalf) [inline src/classes/job-scheduler.ts:74]: what about doing:
`const { startDate, ...filteredRepeatOpts } = repeatOpts;`
in this line instead of in line 108

**Review by fgozdz — COMMENTED** [review]
with the changes suggested by Rogger it looks good

File: tests/test_job_scheduler.ts (hunk 2 above)
Comment (manast) [reply tests/test_job_scheduler.ts:750]: In fact it can be removed completely.

**Pull request merged**